### PR TITLE
scrape: Move tests to testutil

### PR DESCRIPTION
**Pull request by dzeban**
Signed-off-by: Alex Dzyoba <alex@dzyoba.com>

### CODE REVIEW

File: scrape/scrape_test.go [hunk 1]
```diff
@@ -196,17 +195,11 @@ func TestScrapePoolStop(t *testing.T) {
 	}
 
 	mtx.Lock()
-	if len(stopped) != numTargets {
-		t.Fatalf("Expected 20 stopped loops, got %d", len(stopped))
-	}
+	testutil.Equals(t, len(stopped), numTargets)
```
Comment (cstyan) [inline scrape/scrape_test.go]: Can you include the message here so we know which value was expected vs what the output was in the test, you can include a string after `numTargets`

Comment (dzeban) [reply scrape/scrape_test.go]: Good point, thanks! I've update the PR.

File: scrape/scrape_test.go [hunk 2]
```diff
@@ -744,15 +716,10 @@ func TestScrapeLoopRunCreatesStaleMarkersOnFailedScrape(t *testing.T) {
...
-	if appender.result[0].v != 42.0 {
-		t.Fatalf("Appended first sample not as expected. Wanted: %f Got: %f", appender.result[0].v, 42.0)
-	}
-	if !value.IsStaleNaN(appender.result[6].v) {
-		t.Fatalf("Appended second sample not as expected. Wanted: stale NaN Got: %x", math.Float64bits(appender.result[5].v))
-	}
+	testutil.Equals(t, 27, len(appender.result), "Appended samples not as expected")
+	testutil.Equals(t, 42.0, appender.result[0].v, "Appended first sample not as expected")
+	testutil.Assert(t, value.IsStaleNaN(appender.result[6].v),
+		"Appended second sample not as expected. Wanted: stale NaN Got: %x", math.Float64bits(appender.result[5].v))
```
Comment (csmarchbanks) [inline scrape/scrape_test.go]: Possibly a bug in the previous code, but shouldn't both be `result[6]`?

File: scrape/scrape_test.go [hunk 3]
```diff
@@ -806,25 +773,19 @@ func TestScrapeLoopRunCreatesStaleMarkersOnParseFailure(t *testing.T) {
...
-	if appender.result[0].v != 42.0 {
-		t.Fatalf("Appended first sample not as expected. Wanted: %f Got: %f", appender.result[0].v, 42.0)
-	}
-	if !value.IsStaleNaN(appender.result[6].v) {
-		t.Fatalf("Appended second sample not as expected. Wanted: stale NaN Got: %x", math.Float64bits(appender.result[6].v))
-	}
+	testutil.Equals(t, 17, len(appender.result), "Appended samples not as expected")
+	testutil.Equals(t, 42.0, appender.result[0].v, "Appended first sample not as expected")
+	testutil.Assert(t, value.IsStaleNaN(appender.result[6].v),
+		"Appended second sample not as expected. Wanted: stale NaN Got: %x", math.Float64bits(appender.result[5].v))
```
Comment (csmarchbanks) [inline scrape/scrape_test.go]: Should be `result[6]` here.

File: scrape/scrape_test.go (hunk 2 above)
Comment (dzeban) [reply scrape/scrape_test.go]: Yes, sure. Fixed that.

File: scrape/scrape_test.go (hunk 3 above)
Comment (dzeban) [reply scrape/scrape_test.go]: Also fixed.

**Pull request merged**